main.go

```diff
@@ -29,15 +38,116 @@ func (cfg Config) validateEnvs() error {
 		}
 	}
 
-	if err := input.ValidateIfNotEmpty(string(cfg.Password)); err != nil {
-		if err := input.ValidateIfNotEmpty(string(cfg.AppPassword)); err != nil {
-			return fmt.Errorf("neither password nor app_password is provided")
+	switch isJWTAuthType, isAppleIDAuthType := (cfg.APIKeyPath != "" || cfg.APIIssuer != ""), (cfg.AppPassword != "" || cfg.Password != "" || cfg.ItunesConnectUser != ""); {
+
+	case isAppleIDAuthType == isJWTAuthType:
+
+		return fmt.Errorf("one type of authentication required, either provide Apple ID with password/app password or API key with issuer")
+
+	case isAppleIDAuthType:
+
+		if err := input.ValidateIfNotEmpty(string(cfg.ItunesConnectUser)); err != nil {
+			return fmt.Errorf("no Apple ID provided")
+		}
+		if err := input.ValidateIfNotEmpty(string(cfg.Password)); err != nil {
+			if err := input.ValidateIfNotEmpty(string(cfg.AppPassword)); err != nil {
+				return fmt.Errorf("neither password nor app_password is provided")
+			}
+		}
+
+	case isJWTAuthType:
+
+		if err := input.ValidateIfNotEmpty(string(cfg.APIIssuer)); err != nil {
+			return fmt.Errorf("no API Issuer provided")
 		}
+		if err := input.ValidateIfNotEmpty(string(cfg.APIKeyPath)); err != nil {
+			return fmt.Errorf("no API Key path provided")
+		}
+
 	}
 
 	return nil
 }
 
+// prepares key and returns the ID the tool need to use
+func prepareAPIKey(apiKeyPath string) (string, error) {
+	// input should accept url and filepath
+	// 4 location existing, check the files there first
+	// if there is a file on any of the 4 locations use that
+	// if ther eis no that file found then move it to one of the locations
+	// if the filename is not conventional for the tool then create a new unique one
+
+	// see these in the altool's man page
+	var (
+		keyPaths = []string{filepath.Join(os.Getenv("HOME"), ".appstoreconnect/private_keys"), "./private_keys", filepath.Join(os.Getenv("HOME"), "private_keys"), filepath.Join(os.Getenv("HOME"), ".private_keys")}
+		keyID    = "Bitrise" // as default if no ID found in file name
+	)
+
+	// parse string to url
+	fileURL, err := url.Parse(apiKeyPath)
```

Comment:
this could be a function, like `keyName`, can be tested and well-scoped logical block